aten/src/ATen/gen_backend_select_register.py

```diff
@@ -0,0 +1,48 @@
+from code_template import CodeTemplate
+
+FUNCTION_REGISTRATION = CodeTemplate("""\
+.op(torch::RegisterOperators::options()
+  .schema("${schema_string}")
+  .impl_unboxedOnlyKernel<decltype(${function_name}), &${function_name}>(DispatchKey::BackendSelect)
+  .aliasAnalysis(AliasAnalysisKind::FROM_SCHEMA))
+""")
+
+FUNCTION_DEFINITION = CodeTemplate("""\
+Tensor ${function_name}(${method_formals}) {
+  DispatchKey key = options.computeDispatchKey();
+  static auto op = c10::Dispatcher::singleton().findSchemaOrThrow("aten::${name}", "${overload_name}");
+  return op.callUnboxedWithDispatchKey<${formals_types}>(key, ${type_method_actuals});
+}
+""")
+
+def register_backend_select_methods(declarations, template_path, file_manager):
+    backend_select_method_definitions = []
+    backend_select_function_registrations = []
+
+    for decl in declarations:
+        for option in decl["options"]:
+            if '_like' in option['name'] or 'new_' in option['name'] or 'to' in option['name']:
+                continue
+
+            name = option['name']
+            if 'overload_name' in option and option['overload_name'] != '':
```

Comment:
nit: more compact way to test this is `option.get('overload_name')`

Reply:
But I'm a little unsure why these need to be coded defensively as I was under the impression that `overload_name` and `dynamic_type` (below) were guaranteed to be set in the dictionaries.

Reply:
its killed due to a new key 'backend_select' in the native_functions.yaml

Reply:
I have the same question as @ezyang and I don't understand your response.

Reply:
...I'd still change this to `option.get('overload_name', '') != ''` if you have the chance